extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/DefaultAndNamedMongoClientConfigTest.java

```diff
@@ -55,6 +55,10 @@ public void testNamedDataSourceInjection() {
         assertThat(Arc.container().instance(MongoClient.class, Default.Literal.INSTANCE).get()).isNotNull();
         assertThat(Arc.container().instance(MongoClient.class, NamedLiteral.of("cluster2")).get()).isNotNull();
         assertThat(Arc.container().instance(MongoClient.class, NamedLiteral.of("cluster3")).get()).isNull();
+
+        // assert using a CDI name results in the retrieving the client
+        assertThat(Arc.container().instance("cluster2").get()).isNotNull();
+        assertThat(Arc.container().instance("cluster3").get()).isNull();
```

Comment:
If we need uniqueness for all the types across the CDI container, that won't fly. We need to qualify the name with the type.

Also we should probably provide an API util to generate the names.